### PR TITLE
Add compat for Spin's Auth Pattern

### PR DESCRIPTION
**Pull request by hunterowens**
Would love your thoughts on how to best architect this, currently testing rn.

### CODE REVIEW

**Comment by johnclary** [comment]
IIRC I this one pretty straightforward following [these docs](https://web.spin.pm/datafeeds). We're not bothering with the refresh token and just request a new token every time. Is that bad?

Probably not helpful, but here's where we do auth stuff in our code that actually gets mds data: 
https://github.com/cityofaustin/transportation-dockless-processing/blob/77965ad424d6382149c0fbdca8a4be0f4d788821/get_data.py#L197

**Comment by johnclary** [comment]
Ah I see y'all are mixing in classes for provider-specific auth. Cool. That's more transparent than what we're doing, which is putting the auth params in a `secrets.py` config. Because these are just flavors of oauth with slightly different payload params, our approach is less fuss handling the provider nuances, but it's also hidden because it's in a secret config.

**Comment by johnclary** [comment]
Maybe refactor `OAuthClientCredentialsAuth` to handle an arbitrary payload, and define the provider-specific auth payloads as mix ins?

**Comment by hunterowens** [comment]
yeah. I dunno. @thekaveman do you have strong feelings? I suspect John and I feel this pain point more with 8+ operators vs 4. Happy to architect and fix as needed.

File: mds/api/client.py [hunk 1]
```diff
@@ -34,6 +34,8 @@ def _auth_session(self, provider):
         if hasattr(provider, "token") and not hasattr(provider, "token_url"):
             # auth token defined by provider
             return self.auth_token_session(provider)
+        elif hasattr(provider, "email"):
```
Comment (hunterowens) [inline mds/api/client.py]: This seems brittle, do something like `provider == spin`

Comment (hunterowens) [reply mds/api/client.py]: Fixed, thanks @ian-r-rose

**Comment by thekaveman** [comment]
I'm OK with an approach like this until we figure out a more permanent solution.

You sure this works though? I don't see where that mixin is.. [mixed in](https://github.com/CityofSantaMonica/mds-provider/pull/74/files#diff-19f96ea08c351382a35fbf93425ce85aR13)?

**Pull request merged**